plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/FontUtil.java

```diff
@@ -54,4 +84,7 @@ public void widgetDisposed(DisposeEvent event) {
     });
   }
 
+  private static boolean hasStyle(FontData fontData, int style) {
+    return (fontData.getStyle() & style) != 0;
```

Comment:
Gotcha! Yeah, they shouldn't overlap. But e.g. if we create a new method that makes the font bold and italic and we pass in those bits in a single int, that could cause a problem similar to @briandealwis 's example.